### PR TITLE
render: Add DrawCommand::CubicCurveTo

### PR DESCRIPTION
**Pull request by Dinnerbone**
This is the first DrawCommand that isn't available in the swf format. It's exposed to AS3 through `Graphics.cubicCurveTo` and `Graphics.drawPath`.

To avoid confusion, the existing `CurveTo` has been renamed to `QuadraticCurveTo`.

This is a hard dependency for device rendering support - a lot of fonts use cubic curves, not just quadratic curves.

For hit testing I just used lyon to generate the bounding box, and to split it to quadratic curves. I think I'd propose we switch all other hit testing to lyon:
- It's likely faster than our implementation, they have a lot more short-circuits and fast-approximations than we do. Not tested to verify this though.
- They have a *lot* more tests than we do :D

### CODE REVIEW

File: render/src/shape_utils.rs
Comment (moulins) [inline render/src/shape_utils.rs:700]: Nit: the constant `0.01` is used twice, so it should probably be given a name.

File: render/src/shape_utils.rs
Comment (Dinnerbone) [reply render/src/shape_utils.rs]: Good idea, done!

File: render/src/shape_utils.rs
Comment (Dinnerbone) [reply render/src/shape_utils.rs]: Done!

File: render/src/shape_utils.rs
Comment (torokati44) [inline render/src/shape_utils.rs:1319]: it's somewhat strange that `Rectangle<Coordinate>` doesn't have a `grow` or `expand` method for this, but alas, it doesn't.

Comment (Dinnerbone) [reply render/src/shape_utils.rs:1319]: We have so many pseudo math/geom types that all have different capabilities. I'd love to just make a ruffle-geom crate and get it over with... One day 😅

File: render/src/shape_utils.rs
Comment (torokati44) [inline render/src/shape_utils.rs:770]: A comment here would be nice that this "doesn't make sense", but is for performance, in lieu of an early break from the `for_each_quadratic_bezier` - but this is such a minor nit...

File: render/src/shape_utils.rs
Comment (torokati44) [inline render/src/shape_utils.rs:1305]: Honestly, I can't make up my mind on what to think about this... Many pros and cons on both sides...

File: render/src/shape_utils.rs
Comment (torokati44) [inline render/src/shape_utils.rs:6]: again, a minor nit: `... into quadratic ones` ?

Comment (Dinnerbone) [reply render/src/shape_utils.rs:6]: Feels a little tautological to me 😅

Comment (torokati44) [reply render/src/shape_utils.rs:6]: Also maybe a unit for this number, or "meaning of the magnitude" would be nice to mention, but not critical.

Comment (torokati44) [reply render/src/shape_utils.rs:6]: > Feels a little tautological to me 😅
Holy moly, I forgot the middle of the sentence by the time I got to its end... Why am _I_ code reviewing _after midnight_?

**Comment by torokati44** [comment]
Huh, did none of the ported ACID tests get fixed by this?

**Review by torokati44 — APPROVED** [review]

**Comment by torokati44** [comment]
Left some review notes, nothing worthy to block the merge, I'll leave it up to you @Dinnerbone, whether you make any further changes based on them.

**Comment by Dinnerbone** [comment]
They don't use it. I haven't actually seen any content online that uses it, outside of blog posts saying "you should use this!"

**Pull request merged**
